### PR TITLE
Fixing LinalgTransform bug

### PR DESCRIPTION
**Pull request by giuseros**
This is fixing the bug I raised in https://github.com/google/iree-llvm-sandbox/issues/300 and adding a test case

### CODE REVIEW

**Comment by giuseros** [comment]
cc @ftynse

File: lib/Dialects/LinalgTransform/IR/TransformOpInterface.cpp [hunk 1]
```diff
@@ -79,12 +79,15 @@ transform::TransformState::applyTransform(TransformOpInterface transform) {
   if (failed(transform.apply(results, *this)))
     return failure();
 
-  for (auto en : llvm::enumerate(transform->getResults()))
+  for (Value target : transform->getOperands()) {
```
Comment (ftynse) [inline lib/Dialects/LinalgTransform/IR/TransformOpInterface.cpp:82]: Nit: drop trivial braces.

File: test/LinalgTransform/double_tiling_alp.mlir [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// RUN: mlir-proto-opt -linalg-interp-transforms %s | FileCheck %s
+
+// CHECK-LABEL: func @matmul_tensors(
+// CHECK-SAME:    %[[TA:[0-9a-z]+]]: tensor<128x128xf32>
+// CHECK-SAME:    %[[TB:[0-9a-z]+]]: tensor<128x128xf32>
+// CHECK-SAME:    %[[TC:[0-9a-z]+]]: tensor<128x128xf32>
+// CHECK-SAME:  -> tensor<128x128xf32> {
```
Comment (ftynse) [inline test/LinalgTransform/double_tiling_alp.mlir:7]: I'm not sure these check lines are necessary. The test seems to ensure the interpreter doesn't assert, so just having one label line and ensure some IR was produced is enough, the rest will just create churn should the syntax change in the future. Also consider adding a simple comment stating that only the success of the transformation is relevant here.

**Review by ftynse — APPROVED** [review]
LGTM with changes

**Comment by giuseros** [comment]
@ftynse , all the changes have been applied (sorry for the delay)

**Pull request merged**